Fix #1889: Proper dimensions for bottom legend on line chart

KidA78 opened the pull request:
Fix includes:
- Accounting for chart X axis on legend bottom position
- Adjusts available height according to adjusted bottom margin
- Added example of line chart w/ bottom margin

Usage:
```
        chart = nv.models.lineChart().legendPosition("bottom");
```

liquidpele commented:
I'd prefer not to merge in the example html... you could include this option in the existing example html page somehow, but I'd like to avoid having too many example pages as it becomes kind of difficult to maintain and browse.

KidA78 commented:
@liquidpele I hear you.  Just made the change.  Removed the extra example I added, and put in a toggle button to toggle the legend between top and bottom. Give it a shot and let me know what you think

liquidpele commented on examples/lineChart.html:
```diff
@@ -29,6 +29,8 @@
 <div style="position:absolute; top: 0; left: 0;">
     <button onclick="randomizeFillOpacity();">Randomize fill opacity</button>
     <button onclick="expandLegend();">Expand/Contract Legend</button>
+    <button onclick="toggleLegend();">Toggle Legend Top/Bottom</button>
```
This is going to make the buttons overlap on the legend when it's at the top, at least in the default layout where it's in an iframe with a set width... but that's an issue that's beyond the scope of this PR so it's fine for now.

KidA78 commented:
@liquidpele Thanks!
